### ISSUE TITLE
Printer-friendly Version of the Guides

### ISSUE THREAD

**Issue by posixeleni** [IQSS/dataverse#2148]
discussed with @eaquigley that we need to make available a printer-friendly version of the guides.

@cchoirat confirmed with me that each time we make an HTML version of the Sphinx docs we would need to also do:

```
make latexpdf
```

or 

```
make pdflatex
```

This would need to be done by development and then in the guides we would ideally have a "View Printer Friendly Version" option on each page.

**Comment by pdurbin** [comment]
> This would need to be done by development

We would probably automate this with Jenkins as part of a normal build, which already makes HTML versions of the guides.

Code contributors such as @bencomp have expressed interest in how our Jenkins jobs are set up (#1936) so we should probably consider committing some scripts used by Jenkins (such as the oneliners we use to build docs) into the code base itself.

**Comment by cchoirat** [comment]
make latexpdf

**Comment by bencomp** [comment]
As long as someone is adding this to the Jenkins job, could they also add `make epub`?

**Comment by pdurbin** [comment]
Today @cchoirat mentioned that epub versions should be possible.

**Comment by cchoirat** [comment]
Indeed, just as @bencomp wrote: `make epub`.  It's supported out of the box.

**Comment by pdurbin** [comment]
Wow, `make epub` seems to work great. @bencomp and I were checking it out: http://irclog.iq.harvard.edu/dataverse/2015-05-20#i_20046

**Comment by pdurbin** [comment]
@scolapasta and I agreed that we'd use this ticket to track the addition of "last updated" (suggested by @posixeleni ) to the guides so people can see which version they are looking at.

In bb9b0cb I just uncommented `html_last_updated_fmt` to enable the default formatting, so "Last updated on May 26, 2015." was added to the bottom of http://guides.dataverse.org/en/4.0.1/user/appendix.html (I just ran a build). You can compare this to the version with no date at http://guides.dataverse.org/en/4.0/user/appendix.html

@posixeleni does this look ok for the HTML version, at least? I believe this is what you were suggesting.

I can only assume we can add "last updated" to PDFs and ePubs as well (maybe @cchoirat knows). Probably we don't need this on every page though! :)

**Comment by posixeleni** [comment]
Thanks @pdurbin.

Is this the "Last Update" date for the entire guide or just this particular page of the guide? The text looks great I just worry that it is lost a little bit in the footer text. Is there any way to bump this "last update" text to show closer to the actual guide text?

**Comment by pdurbin** [comment]
@posixeleni if you click around you'll see the same date on the bottom of every page.

I haven't worked on templates or themes or whatever they are called in Sphinx but maybe @mheppler or @raprasad or @cchoirat or others can comment on how easy it is to move the date from `html_last_updated_fmt` out of the footer to somewhere else on the page.

**Comment by mheppler** [comment]
I can look into reviewing where the date is displayed in the footer. I'd also like to see why there are two Sphinx links as well.

@pdurbin, if you have time to meet today, let's take this discussion offline and review some options before pushing anymore changes.

**Comment by mheppler** [comment]
- Talked to @pdurbin, moved this ticket into "Dev".
- Decided that the date made more sense in the footer than higher up, in the context of the page, because the date reflects changes to the entire guides site and not to an individual page.
- Added guides version number.
- Removed duplicate Sphinx link.
- See attached screenshot for mocked up layout revisions that I am checking in shortly.
![screen shot 2015-05-26 at 1 56 33 pm](https://cloud.githubusercontent.com/assets/687227/7819350/23d9fa50-03af-11e5-9a99-b40218df6f05.png)

**Comment by posixeleni** [comment]
:+1:

**Comment by mheppler** [comment]
Cleaned up layout of Guides footer, to show new version number and last updated date. Also removed redundant Sphinx link.

@pdurbin, the layout changes are committed. Let me know if you need anything else.

**Comment by pdurbin** [comment]
@mheppler as we discussed I'd probably like some input from you and others on where to link to PDF and ePub versions of the docs. There's a similar ticket for Zelig: https://github.com/IQSS/docs.zeligproject.org/issues/5

Building ePubs is easy and I reconfigured Jenkins to build them. You can find the "next" (unreleased) version here: http://guides.dataverse.org/en/next/Dataverse.epub

I don't believe we've built PDFs on the Jenkins server itself (build.hmdc) before and it isn't working. (We build a PDF for Zelig but on the machinne hosting http://docs.zeligproject.org itself.) Here's what I know:

- I'm not sure what to install on my Mac to get `make latexpdf` working. It seems to be looking for a binary called `pdflatex`
- I **can** build a PDF within a Vagrant VM spun up from https://github.com/IQSS/zeligproject.org
  - Here's how it looks (I had to add a logo in a6687cf to get it to build): http://dvn-vm1.hmdc.harvard.edu/tmp/Dataverse.pdf
- Jenkins (build.hmdc) can't build PDFs yet. Here are the logs for a build failure: https://build.hmdc.harvard.edu:8443/job/docs.dataverse.org/828/console

I suspect that some of the config changes mentioned here might need to be applied to the Jenkins server (build.hmdc) before it can create PDFs: https://github.com/IQSS/zeligproject.org/commit/903bd149b39b64ee851c42c2ff374915700c9b06

**Comment by pdurbin** [comment]
Another option for generating PDFs (rather than messing with the Jenkins server, build.hmdc) would be to switch to readthedocs. It seems to be a supported feature: https://docs.readthedocs.org/en/latest/features.html#pdf-generation

**Comment by mheppler** [comment]
For the PDF and ePub links, I'd suggest that we add them to the footer. I don't think that they make much sense in the sidebar (as suggested in that Zelig Project issue) or elsewhere, unless we build an entire standalone page, with a heading, some help text, and links to the various versions.

Here is a mockup of the links in the footer, which I believe makes the most sense.

![screen shot 2015-05-26 at 4 35 37 pm](https://cloud.githubusercontent.com/assets/687227/7822736/9776531c-03c5-11e5-97db-27182f807442.png)

**Comment by posixeleni** [comment]
:+1:

**Comment by pdurbin** [comment]
> Here's how it looks (I had to add a logo in a6687cf to get it to build): http://dvn-vm1.hmdc.harvard.edu/tmp/Dataverse.pdf

Hmm, I just opened the PDF and noticed that "How the Guides Are Organized" appears on page 79 (chapter 5) when I believe we want that text to be very early, probably just after the table of contents.

@posixeleni would you be interested in experimenting with https://readthedocs.org in a test repo to figure out some of these PDF layout issues?

**Comment by posixeleni** [comment]
@pdurbin How the Guides Are Organized should definitely be Chapter 1 this is the root level index.rst file so I dont know why it's put it as Chapter 5? @cchoirat @raprasad would either of you know why this would be happening?

**Comment by pdurbin** [comment]
@doprdele [hooked me up](https://help.hmdc.harvard.edu/Ticket/Display.html?id=216558) with the `scl enable python27` magic to get the Jenkins server to build the PDFs:

`cd $WORKSPACE/doc/sphinx-guides; make clean; scl enable python27 "make html";  scl enable python27 "make latexpdf"; cp build/latex/Dataverse.pdf build/html; scl enable python27 "make epub"; cp build/epub/Dataverse.epub build/html; cd build/html; tar cvzf /tmp/guides.tgz .`

That means that both PDFs and ePubs can be now be viewed here:

- http://guides.dataverse.org/en/next/Dataverse.pdf
- http://guides.dataverse.org/en/next/Dataverse.epub

Again "next" means "next unreleased version". As before, the Jenkins job polls the git repo once an hour and does a build into this "next" directory (which will become 4.0.1 as of this writing).

Unfortunately, the PDF especially looks pretty rough. We might do well do to seek advice from @cchoirat since the [Zelig PDF](http://docs.zeligproject.org/en/latest/Zelig.pdf) looks so much nicer. Perhaps she can advise on what should be installed on a Mac to build the PDF in the first place. (I've only been able to build the PDF on Linux.)

All this is to say that there's still more work before PDFs and ePubs are of shippable quality. From talking to @scolapasta and @eaquigley I'll move this issue to the "In Design" milestone and remove my assignment.

**Comment by eaquigley** [comment]
@pdurbin you mentioned earlier today there are a lot of blank pages in the docs. i took a look at the zelig docs and they have the same amount of blank pages so i am presuming this is a sphinx thing.

**Comment by pdurbin** [comment]
@eaquigley did I? Well, I'm way less worried about blanks pages than I am about weirdness like this:

Lines cut off (code examples too):

![dataverse pdf_-_2015-05-27_18 44 35](https://cloud.githubusercontent.com/assets/21006/7849190/d9f8aef2-04a0-11e5-9ec9-a50f266f65c9.png)

Image placement and w e i r d  s p a c i n g:

![dataverse pdf_-_2015-05-27_18 45 15](https://cloud.githubusercontent.com/assets/21006/7849195/de267766-04a0-11e5-9d0e-b8b19edc1e25.png)

And there's the misplacement of "how the guides are organized" I mentioned at https://github.com/IQSS/dataverse/issues/2148#issuecomment-105918922

Basically, the PDF is a bit of a mess. :(

At least the ePub looks pretty good! :)

**Comment by eaquigley** [comment]
Need to set up a meeting to discuss a few things with guides. For example, moving Dataverse guides to a different repo so others can easily fork it and make their own documentation.

**Comment by djbrooke** [comment]
This is available on readthedocs in case anyone wants to print or download the guides. 

https://readthedocs.org/projects/dataverse-guides/downloads/pdf/latest/
